controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -490,7 +599,7 @@ public String getScopeName() {
 
     @Override
     public CompletableFuture<TxnStatus> commitTransaction(final int epoch, final UUID txId) {
-        CompletableFuture<TxnStatus> future = verifyLegalState(checkTransactionStatus(epoch, txId).thenApply(x -> {
+        CompletableFuture<TxnStatus> future = verifyLegalState(() -> checkTransactionStatus(epoch, txId).thenApply(x -> {
```

Comment:
This method shall be called only if the transaction is created and then subsequently sealed. Do we really need `verifyLegalState` check here. I think it is redundant.